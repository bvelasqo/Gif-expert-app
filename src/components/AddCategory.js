import React, {useState} from "react";
import PropTypes from "prop-types";

const AddCategory = ({setCategories}) => {
	const [inputValue, setInputValue] = useState("");

	const handleInputChange = (event) => {
		setInputValue(event.target.value);
	};

	const handleSubmit = (event) => {
		event.preventDefault();
		if (inputValue.trim().length > 2) {
			setCategories((categories) => [inputValue, ...categories]);
			setInputValue("");
		} else console.log("The category must have at least 3 characters");
	};

	return (
		<form onSubmit={handleSubmit}>
			<div>
				<div className="container-input">
					<input
						type="text"
						value={inputValue}
						placeholder="Add a category"
						onChange={handleInputChange}
					/>
				</div>
				<hr style={{width: "67%", borderColor:"white"}}/>
			</div>
		</form>
	);
};

AddCategory.prototype = {
	setCategories: PropTypes.func.isRequired,
};

export {AddCategory};
