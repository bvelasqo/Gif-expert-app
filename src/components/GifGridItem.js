import React from "react";
import PropTypes from "prop-types";

const GifGridItem = ({id, title, url}) => {
	return (
		<div className="gif_item animate__animated animate__zoomIn">
			<img
				key={id}
				src={url}
				alt={title}
				className="gif animate__animated animate__zoomIn"
			/>
			<p className="title_image animate__animated animate__zoomIn animate__delay">{title}</p>
		</div>
	);
};

GifGridItem.propTypes = {
  id: PropTypes.string,
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export {GifGridItem};