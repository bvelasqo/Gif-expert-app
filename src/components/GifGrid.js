import React from "react";
import PropTypes from "prop-types";
import {useFetchGifs} from "../hooks/useFetchGifs";
import {GifGridItem} from "./GifGridItem";

const GifGrid = ({category}) => {
	const {data: gifs, loading} = useFetchGifs(category);

	return (
		<div className="main_gif_grid">
			<h3 className="category animate__animated animate__bounceIn">Category: {category}</h3>
			{/* loading con && */}
			{loading && <p>Loading...</p>}
			{/* loading con || */}
			{loading ? (
				<div className="animate__animated animate__flash">
					<img
						src="https://media.giphy.com/media/3oEjI6SIIHBdRxXI40/giphy.gif"
						alt="loading"
					/>
				</div>
			) : (
				<div className="gif_grid">
					{gifs.map((image) => (
						<>
							<GifGridItem key={image.id} {...image} />
						</>
					))}
				</div>
			)}
		</div>
	);
};

GifGrid.prototype = {
	category: PropTypes.string.isRequired,
};

export default GifGrid;
