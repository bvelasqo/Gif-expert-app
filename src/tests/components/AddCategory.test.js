import {shallow} from "enzyme";
import "@testing-library/jest-dom"
import {AddCategory} from "../../components/AddCategory";
describe("AddCategory", () => {

	const setCategories = jest.fn();
	let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();

  });

	test("Should to change the input value", () => {
		let input = wrapper.find("input");
		const value = "Dragon Ball";
    // simulate() is a method from enzyme that allows us to simulate an event (see https://enzymejs.github.io/enzyme/docs/api/ShallowWrapper/simulate.html)
    // change is an event that we can simulate (see https://reactjs.org/docs/events.html)
    // target is a property that we can use to get the value of the input (see https://reactjs.org/docs/events.html#event-pooling)
		input.simulate("change", {target: {value}});
		input = wrapper.find("input");
		expect(input.prop("value")).toBe(value);
	});


  test("Should not call setCategories when form is submitted and input is empty", () => {
    const form = wrapper.find("form");
    form.simulate("submit", {preventDefault() {}});
    expect(setCategories).not.toHaveBeenCalled();
  });

  test("Should call setCategories when form is submitted and input is not empty", () => {
    const value = "Dragon Ball";
    wrapper.find("input").simulate("change", {target: {value}});
    const form = wrapper.find("form");
    form.simulate("submit", {preventDefault() {}});
    const input = wrapper.find("input");
    expect(setCategories).toHaveBeenCalled();
    expect(input.prop("value")).toBe("");
  });
});
