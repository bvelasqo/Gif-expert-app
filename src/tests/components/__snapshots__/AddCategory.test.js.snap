// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Add a category"
    type="text"
    value=""
  />
</form>
`;
